resource/identity_upgrader.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package resource
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-framework/resource/identityschema"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+// Implementation handler for an UpgradeIdentity operation.
+//
+// This is used to encapsulate all upgrade logic from a prior identity to the
+// current version when a Resource implements the
+// ResourceWithUpgradeIdentity interface.
+type IdentityUpgrader struct {
+	// Schema information for the prior identity version. While not required,
+	// setting this will populate the UpgradeResourceIdentityRequest type Identity
+	// field similar to other Resource data types. This allows for easier data
+	// handling such as calling Get() or GetAttribute().
+	//
+	// If not set, prior identity data is available in the
+	// UpgradeResourceIdentityRequest type RawIdentity field.
+	PriorSchema *identityschema.Schema
+
+	// Version is the version schema that this Upgrader will handle, converting
+	// it to Version+1.
+	Version int64
+
+	// Type describes the schema that this function can upgrade.
+	Type tftypes.Type
+
```

Comment:
I don't see either of these used anywhere, are they needed here? (version should be defined in the key of the map of identity upgraders)

```suggestion
```